core/src/postgres.rs

```diff
@@ -315,10 +313,22 @@ impl TableProviderFactory for PostgresTableProviderFactory {
         #[cfg(feature = "postgres-federation")]
         let read_provider = Arc::new(read_provider.create_federated_table_provider()?);
 
+        // Extract PostgresWriteConfig from session or use defaults
+        let write_config = state
+            .config()
+            .get_extension::<write::PostgresWriteConfig>()
+            .map(|ext| (*ext).clone())
+            .unwrap_or_else(|| write::PostgresWriteConfig {
+                batch_flush_interval: Duration::from_secs(1),
+                batch_size: 1_000_000,
+                num_records_before_stop: None,
+            });
```

Comment:
It's weird that the default values are different from the values defined above in the `read_write_table_provider` method... Should they be different?

Also, it would be nice to implement a Default trait for PostgresWriteConfig and move the defaults there.

Reply:
fixed in dfe7a11c6937f93b85870df63727fba077ff5233